### DISCUSSION TITLE
Common flows for Android and iOS or separation of flows?

### DISCUSSION THREAD

**Discussion by jbouche-work** [mobile-dev-inc/Maestro#2206] [General]
Hi,

Do you recommand to have common flows for Android and iOS with conditions ?
Flows :
-- common-test-feature-x.yaml
-- common-test-feature-y.yaml
-- common-test-feature-z.yaml

Or do you recommand to create specific flows for iOS and specific flows for Android ?
Flows :
-- Android
----  android-test-feature-x.yaml
----  android-test-feature-y.yaml
----  android-test-feature-z.yaml
-- iOS
---- ios-test-feature-x.yaml
---- ios-test-feature-y.yaml
---- ios-test-feature-z.yaml

Our mobile app has some minor differences between the iOS version and the Android version.

Jérôme.

**Comment by Fishbowler** [comment]
Rubbish answer, but "it depends".

How much platform-specific stuff is there gonna be in the flow? It all comes down to readability / maintainability trade-off. Harder to maintain multiple files. Harder to maintain a single file that's hard to understand.
